Here is the summary of failures from `testing.postgres`, the only package failing:

```
=========================== short test summary info ============================
FAILED tests/test_postgresql.py::TestPostgresql::test_skipIfNotFound_found - ...
FAILED tests/test_postgresql.py::TestPostgresql::test_skipIfNotFound_notfound
FAILED tests/test_postgresql.py::TestPostgresql::test_skipIfNotInstalled_found
FAILED tests/test_postgresql.py::TestPostgresql::test_skipIfNotInstalled_notfound
FAILED tests/test_postgresql.py::TestPostgresql::test_skipIfNotInstalled_with_args_found
FAILED tests/test_postgresql.py::TestPostgresql::test_skipIfNotInstalled_with_args_notfound
```

In summary, all of them are `AttributeError: module 'collections' has no attribute 'Callable'`. This seems to be related with the lack of compatibility with Python 3.10, which makes sense as the library that could not be built hasn't been updated since 2017.

--- a/pkgs/development/python-modules/asn1crypto/default.nix
+++ b/pkgs/development/python-modules/asn1crypto/default.nix
@@ -1,19 +1,25 @@
 { lib
 , buildPythonPackage
-, fetchPypi
+, fetchFromGitHub
+, pytestCheckHook
 }:
 
 buildPythonPackage rec {
   pname = "asn1crypto";
-  version = "1.4.0";
+  version = "1.5.1";
+  format = "setuptools";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "f4f6e119474e58e04a2b1af817eb585b4fd72bdd89b998624712b5c99be7641c";
+  # Pulling from Github to run tests
+  src = fetchFromGitHub {
+    owner = "wbond";
+    repo = "asn1crypto";
+    rev = version;
+    sha256 = "sha256-M8vASxhaJPgkiTrAckxz7gk/QHkrFlNz7fFbnLEBT+M=";
   };
 
-  # No tests included
-  doCheck = false;
+  checkInputs = [
+    pytestCheckHook
+  ];
 
   meta = {
     description = "Fast ASN.1 parser and serializer with definitions for private keys, public keys, certificates, CRL, OCSP, CMS, PKCS#3, PKCS#7, PKCS#8, PKCS#12, PKCS#5, X.509 and TSP";

--- a/pkgs/development/python-modules/testing-common-database/default.nix
+++ b/pkgs/development/python-modules/testing-common-database/default.nix
@@ -9,6 +9,11 @@ buildPythonPackage rec {
     sha256 = "0wvdv0frl7xib05sixjv9m6jywaa2wdhdhsqqdfk45akk2r80pcn";
   };
 
+  postPatch =  ''
+    substituteInPlace src/testing/common/database.py \
+      --replace "collections.Callable" "collections.abc.Callable"
+  '';
+
   # There are no unit tests
   doCheck = false;
 

